src/unused_finder/export_collector.rs

```diff
@@ -0,0 +1,477 @@
+use std::{
+    collections::{HashMap, HashSet},
+    iter::FromIterator,
+    sync::Arc,
+};
+
+use swc_core::{
+    common::{
+        comments::{CommentKind, Comments, SingleThreadedComments},
+        BytePos, Spanned,
+    },
+    ecma::{
+        ast::{
+            BindingIdent, CallExpr, Callee, Decl, ExportAll, ExportDecl, ExportDefaultDecl,
+            ExportDefaultExpr, ExportSpecifier, Id, ImportDecl, ImportSpecifier, Lit,
+            ModuleExportName, NamedExport, Pat, Str, TsImportEqualsDecl,
+        },
+        visit::{Visit, VisitWith},
+    },
+};
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+pub enum ExportedItem {
+    Named(String),
+    Default,
+    Namespace,
+    ExecutionOnly, // in case of `import './foo';` this executes code in file but imports nothing
+}
+
+impl From<&ImportedItem> for ExportedItem {
+    fn from(i: &ImportedItem) -> Self {
+        match i {
+            ImportedItem::Named(named) => ExportedItem::Named(named.clone()),
+            ImportedItem::Default => ExportedItem::Default,
+            ImportedItem::Namespace => ExportedItem::Namespace,
+            ImportedItem::ExecutionOnly => ExportedItem::ExecutionOnly,
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+pub enum ImportedItem {
+    Named(String),
+    Default,
+    Namespace,
+    ExecutionOnly, // in case of `import './foo';` this executes code in file but imports nothing
+}
+
+impl From<&ExportedItem> for ImportedItem {
+    fn from(e: &ExportedItem) -> Self {
+        match e {
+            ExportedItem::Named(name) => ImportedItem::Named(name.clone()),
+            ExportedItem::Default => ImportedItem::Default,
+            ExportedItem::Namespace => ImportedItem::Namespace,
+            ExportedItem::ExecutionOnly => ImportedItem::ExecutionOnly,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct ExportsCollector {
```

Comment:
its hard to tell whats going on here when this entire file is considered net new code because of the rename :/